session-replay/Sources/DatadogSessionReplay/Drafts/SessionReplayFeature.swift

```diff
@@ -22,30 +22,35 @@ internal class SessionReplayFeature: DatadogFeature, SessionReplayController {
 
     // MARK: - Main Components
 
-    let recorder: Recorder
+    private let recorder: Recording
+    private let processor: Processing
+    private let writer: Writing
 
     // MARK: - Initialization
 
-    convenience init(configuration: SessionReplayConfiguration) {
-        let rumContextReceiver = RUMContextReceiver()
-        let recorder = Recorder(
-            configuration: configuration,
-            rumContextObserver: rumContextReceiver,
-            processor: Processor()
+    init(configuration: SessionReplayConfiguration) {
+        let writer = Writer()
+
+        let processor = Processor(
+            queue: BackgroundAsyncQueue(named: "com.datadoghq.session-replay.processor"),
```

Comment:
Just curious if there's a huge performance difference between sync and asynch queue here?